src/kagglehub/exceptions.py

```diff
@@ -12,3 +18,38 @@ class BackendError(Exception):
 
 class DataCorruptionError(Exception):
     pass
+
+
+class KaggleApiHTTPError(requests.HTTPError):
+    def __init__(self, message: str, response: Optional[requests.Response] = None):
+        super().__init__(message, response=response)
+
+
+def kaggle_api_raise_for_status(response: requests.Response):
+    """
+    Wrapper around `response.raise_for_status()` that provides nicer error messages
+    See: https://requests.readthedocs.io/en/latest/api/#requests.Response.raise_for_status
+    """
+    try:
+        response.raise_for_status()
+    except requests.HTTPError as e:
+        message = str(e)
+
+        if response.status_code in {HTTPStatus.UNAUTHORIZED, HTTPStatus.FORBIDDEN}:
```

Comment:
@mohami2000 You should use this in the new post method you created. You should also handle 400 (Bad Request / Invalid Argument) here to show a nicer error message to the user.